pkg/reconciler/route/domains/domains.go

```diff
@@ -54,13 +55,22 @@ func DomainNameFromTemplate(ctx context.Context, r *v1alpha1.Route, name string)
 	domainConfig := config.FromContext(ctx).Domain
 	domain := domainConfig.LookupDomainForLabels(r.ObjectMeta.Labels)
 	annotations := r.ObjectMeta.Annotations
+
+	// Generates an unique string that can be used as part of domain name
+	// The unique string is generated from KnRoute's UID and shortened to 5 symbols
+	var unique string
+	if r.UID != "" {
+		unique = base64.RawURLEncoding.EncodeToString([]byte(r.UID))[:5]
```

Comment:
that's very non-random and bound to collisions.

Reply:
could you please elaborate more? thanks

Reply:
What about using the first few chars of the SHA?

Reply:
I am not sure what are the rules for the SHA (regarding used symbols), that's why I am using `RawURLEncoding()` to be sure there are only symbols suitable for URL.

Reply:
I'm pretty sure sha just uses numbers and letters so you should be ok

Reply:
This also affects the length and can easily trip the k8s name restriction.
Which is very annoying since those errors are not readily surfaced to the service level.

Reply:
This unfortunately does not result in a unique value. The owner reference being used here is a UUID which has uniqueness properties; however, taking only a few characters from it greatly reduces entropy and reduces uniqueness.

Using shorter sequences for uniqueness typically requires tracking the unique ids that have been vended so that you can regenerate an identifier in the case of conflict.

Reply:
In the latest updated, I've used the SHA of the KRoute's UID as @duglin suggested. WDYT?

Reply:
While this is perhaps better, your entropy is still just 20bits, which quite low.
As I asked before:
- can we presume those urls are not for human utilization? 
If so, why not just use the whole md5?

As @markusthoemmes suggested, could you present the issue and the solution for discussion at the API WG (this wednesday 1030PDT, as of now has no agenga).

Reply:
it would be nicer if the result were a bit smaller than an md5 result since we don't want the resulting URLs to be too ugly (just a little ugly :-) ).  I believe docker uses 12 chars instead of 5, and I don't think I've heard people complain about conflict. Perhaps 12 will be ok.